programs/uxd/src/instructions/lsd/borrow_from_lsd_depository.rs

```diff
@@ -0,0 +1,284 @@
+        constraint = user_collateral.mint == collateral_mint.key() @UxdError::InvalidCollateralMint,
+    )]
+    pub user_collateral: Box<Account<'info, TokenAccount>>,
+
+    /// #
+    #[account(
+        mut,
+        constraint = user_redeemable.owner == user.key() @UxdError::InvalidOwner,
+        constraint = user_redeemable.mint == redeemable_mint.key() @UxdError::InvalidRedeemableMint,
+    )]
+    pub user_redeemable: Box<Account<'info, TokenAccount>>,
+
+    /// #
+    /// Tracks the user overall position in the LSD Depository (one per user, per LSD Depository).
+    #[account(
+        init_if_needed,
+        seeds = [depository.key().as_ref(), user.key().as_ref()],
+        space = LSD_POSITION_SPACE,       
+        payer = payer,
+        bump,
+    )]
+    pub lsd_position: AccountLoader<'info, LsdPosition>,
+
+    /// #
+    /// PDA holding the deposited collateral on behalf of the lsd_position
+    #[account(
+        init_if_needed,
+        seeds = [lsd_position.key().as_ref()],
+        token::authority = lsd_position,
+        token::mint = collateral_mint,
+        payer = payer,
+        bump,
+    )]
+    pub lsd_position_collateral: Box<Account<'info, TokenAccount>>,
+
+    /// # price oracle account for the collateral
+    #[account(
+        constraint = collateral_oracle_account.key() == depository.load()?.collateral_oracle_params.oracle_account
+    )]
+    pub collateral_oracle_account: AccountInfo<'info>,
+
+    /// #
+    #[account(mut)]
+    pub redeemable_mint: Box<Account<'info, Mint>>,
+
+    /// #
+    pub collateral_mint: Box<Account<'info, Mint>>,
+
+    /// #
+    pub liquidation_mint: Box<Account<'info, Mint>>,
+
+    /// #
+    pub system_program: Program<'info, System>,
+
+    /// #
+    pub token_program: Program<'info, Token>,
+
+    /// #
+    pub uxd_program: Program<'info, crate::program::Uxd>,
+
+    /// #
+    pub rent: Sysvar<'info, Rent>,
+}
+
+pub(crate) fn handler(ctx: Context<BorrowFromLsdDepository>, collateral_amount: u64, loan_to_value_bps: u16) -> Result<()> {
+    // - 0 [CHECKS] ------------------------------------------------------------
+    {
+        // - 0.1 verify that borrows are enabled
+        let lsd_depository = ctx.accounts.depository.load()?;
+        require!(lsd_depository.borrowing_disabled == false, UxdError::BorrowingDisabled);
+
+    }
+
+    // - 1 [TRANSFER USER'S COLLATERAL] ----------------------------------------
+    {
+        msg!("[transfer] {} collateral from user to his lsd_position",
+            collateral_amount,
+        );
+        token::transfer(
+            ctx.accounts
+                .to_transfer_collateral_from_user_to_lsd_position_collateral_ta_context(),
+            collateral_amount,
+        )?;
+    }
+
+    // - 2 [MINT REDEEMABLES] --------------------------------------------------
+    let redeemable_borrow_amount = collateral_amount / u64::from(loan_to_value_bps) * u64::from(BPS_POWER);
```

Comment:
might want to use checked_math there, i'd feel safer

Reply:
yes need to update to checked everywhere

Reply:
Updating these in the other PR, thanks for the feedback